### DISCUSSION TITLE
Load the static link library

### DISCUSSION THREAD

**Discussion by feng-for** [jnr/jnr-ffi#333] [General]
Hi!
Thank you very much for your help! I want to confirm one thing about JNR. Can JNR load a static Library?

**Comment by headius** [comment]
@feng-for If the library is linked statically but still exports the functions, I think it should work. Do you have an example?

**Comment by basshelal** [comment]
JNR-FFI's backend `jffi` loads native libraries using a thin wrapper around `dlopen` (not sure about the Windows equivalent). I believe this is restricted to [shared libraries](https://en.wikipedia.org/wiki/Shared_library)

* [`dlopen` UNIX spec page](https://pubs.opengroup.org/onlinepubs/009696799/functions/dlopen.html)
* [`dlopen` Linux man page](https://man7.org/linux/man-pages/man3/dlopen.3.html)

Generally speaking, if `dlopen` can open/load it then JNR-FFI can too.
